### ISSUE TITLE
ClassDeclarationSniff bug with readonly classes

### ISSUE THREAD

**Issue by enumag** [squizlabs/PHP_CodeSniffer#3842]
## Describe the bug

The sniff removes a space and makes the code invalid

### Code sample
```php
final readonly class X implements Y {}
```

### To reproduce
1. Create a file called `test.php` with the code sample above...
2. Run `phpcs test.php ...`
3. The sniff changes the code to
```
final readonlyclass X implements Y {}
```

## Expected behavior
The code should be unchanged.

## Versions (please complete the following information)

| | |
|-|-|
| Operating System | debian
| PHP version | 8.2
| PHP_CodeSniffer version | 3.7.2
| Standard | custom
| Install type | composer

## Please confirm:

- [x] I have searched the issue list and am not opening a duplicate issue.
- [x] I confirm that this bug is a bug in PHP_CodeSniffer and not in one of the external standards.
- [ ] I have verified the issue still exists in the `master` branch of PHP_CodeSniffer.

**Comment by fredden** [comment]
@enumag please can you confirm what sniff you think is causing issues? You have mentioned ClassDeclarationSniff, but there are four sniffs named that:
- `PEAR.Classes.ClassDeclaration`
- `PSR1.Classes.ClassDeclaration`
- `PSR2.Classes.ClassDeclaration`
- `Squiz.Classes.ClassDeclaration`

**Comment by enumag** [comment]
Oh I'm sorry, didn't realize there was more than one of them.

```
PHP_CodeSniffer\Standards\Squiz\Sniffs\Classes\ClassDeclarationSniff
```

**Comment by enumag** [comment]
Ah okay. It should work fine when 3.8.0 is released then. Thank you!
